aws-amplify/amplify-js
# feat: PubSub Add Options objects for all Providers and fix: Spelling error and deprecation of old exports

#### Description of changes
This PR covers a bit of housekeeping for the PubSub category. It includes:
- Adding Provider specific Options objects that will provider minor feedback, but don't in any way change the public interface.
- Add deprecation warning for `AWSAppSyncProvider` which has been entirely replaced by `AWSAppSyncRealtimeProvider`
- Fix the spelling of ProvidertOptions and MqttProvidertOptions 
  - Leaving aliases marked deprecated so we don't break anyone using these exports
- Prettier line length remapping to align with the default printWidth of 80

#### Description of how you validated changes
Manual testing exercising features in:
- React app with IoT integration
- React app with DataStore syncing over AppSync
- Angular app with IoT integration

For each app listed above, I verified that the codebase shows deprecation warnings for deprecated resource (mostly to confirm I was pointed at the updated package). 

On the IoT integrated React app, I did deeper testing to verify that any of the ProviderOption interface type works against the IoT Provider, confirming

Unit tests run successfully.

#### Checklist



By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
